docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+fractions of all other terrains on the rest of the land is increased to
+compensate.
+
+The fraction of mountains is directly proportional to the ``steepness``
+parameter. It is about 20% for the default ``steepness`` of 30, and increases
+to 60% for a ``steepness`` of 100. The fractions of all other terrain types are
+reduced when there are lots of mountains, leaving room for plains and grassland.
+
+The ``wetness`` parameter controls directly the amount of forest, which ranges
+between 5% and 35%. An amount of jungle is taken away from this based on the
+fraction of the map covered by hot temperatures, but is not used consistently by
+map generators. ``wetness`` also sets the amount of rivers, ranging from 3% to
+11% of all land including mountains.
+
+The most complicated formulae are for swamp and desert, as they combine
+``wetness`` and ``temperature``. Swamps do not appear at all when the world is
+dry and cold, but their fraction increases with both ``wetness`` and
+``temperature`` --- slightly faster with the former. Deserts may also not appear
+if the world is very cold and wet. Their fraction increases rather quickly with
+``temperature`` and is mildly reduced by ``wetness``.
+
+.. _mapgen-height-generators:
+
+Height-Based Generators
+-----------------------
+
+Three of the available algorithms start by building a height map for the whole
+world: :ref:`mapgen-random` (``RANDOM``), :ref:`mapgen-fractal` (``FRACTAL``),
+and :ref:`Fracture Map <mapgen-fracture>` (``FRACTURE``).
+This height map is then used to assign tiles to continents or seas, and to
+distribute terrains on the map. The algorithm used for this is shared between
+the generators and is described in :ref:`mapgen-terrain-assignment`.
+
+.. _mapgen-random:
+
+Fully Random Height
+^^^^^^^^^^^^^^^^^^^
+
+This generator is extremely simple: it builds a completely random height map and
+smoothes it out.
+Terrain is then assigned to tiles based on their height and temperature as
+described in the :ref:`terrain assignment <mapgen-terrain-assignment>` section
+below, and :ref:`rivers are added on top <mapgen-height-rivers>`.
+
+.. _mapgen-fractal:
+
+Pseudo-Fractal Height
+^^^^^^^^^^^^^^^^^^^^^
+
+This generator works by dividing the map in blocks (five plus the number of
+player islands to be created) and assigning a random height to their
+corners. Each block is then processed recursively, cutting it equally in four
+blocks. The height at the corners of the smaller blocks are computed by
+averaging the heights at the corners of the large blocks and adding a decreasing
+amount of noise. This process is repeated until blocks are one tile wide or
+less.
+
+At the end of the generation, more random noise is added on top of the generated
+height map to give more variety on large maps, with warmer tiles generally
+getting more noise. The generated height map is then used to assign terrains as
+:ref:`described below <mapgen-terrain-assignment>`.
+
```

Comment:
s/terrains/terrain

Instead of "described below" give the section name or something more descriptive.